Makefile

```diff
@@ -1,52 +1,82 @@
-# go-structmapper Makefile
-# This Makefile is not required for using go-structmapper and only
-# provides functionality that is used during development
-
-GO_BUILDFLAGS?=-v
-GO_TEST_FLAGS?=-v
-GO_BENCHMARK_FLAGS?=-run="INVALID" -v
-ROOT_PACKAGE_NAME=github.com/anexia-it/go-structmapper
-
-PACKAGES=$(shell go list ./... | \
-	grep -v '/vendor/' | \
-	sed \
-		-e 's|^$(ROOT_PACKAGE_NAME)/|./|g' \
-		-e 's|^$(ROOT_PACKAGE_NAME)$$|./|g' \
-	)
-SRC_FILES=$(shell find . -name '*.go' | egrep -v '^./vendor/')
+GITTAG ?= $(shell git describe --tags --always)
+GITCOMMIT ?= $(shell git log -1 --pretty=format:"%H")
+GOLDFLAGS ?= -s -w -extldflags '-zrelro -znow' -X go.anx.io/go-anxcloud/pkg/client.version=$(GITTAG)
+GOFLAGS ?= -trimpath
+CGO_ENABLED ?= 0
 
+.PHONY: all
 all: build
 
-format:
-	@echo "Running gofmt..."
-	@gofmt -w -s -r '(a) -> a' $(SRC_FILES)
-	@echo "Running goimports..."
-	@goimports -e=true -w=true $(SRC_FILES)
-
-generate:
-	@echo "Generating..."
-	@go generate $(PACKAGES)
-
-build: format generate
-	@echo "Building packages..."
-	@go install $(GO_BUILDFLAGS) $(PACKAGES)
-
-test: build
-	@echo "Running unit tests..."
-	@rm -rf ./coverage.out
-	@mkdir -p ./coverage.out
-	@$(foreach pkg,$(PACKAGES), \
-		go test $(GO_TEST_FLAGS) -cover -coverprofile \
-			./coverage.out/$(shell echo $(pkg) | \
-		sed -e 's|/|_|g' -e 's|^.||g').coverage $(pkg);)
-	@echo "mode: set" > ./coverage.out/combined
-	@cat ./coverage.out/*.coverage | \
-		sed -e '/mode:.*/d' >> ./coverage.out/combined
-	@go tool cover -html ./coverage.out/combined -o ./coverage.html
-	@go tool cover -func=./coverage.out/combined | grep 'total' | \
-		awk '{print "coverage (all): " $$3 " of statements"; }'
+.PHONY: build
+build: fmtcheck go-lint
+	go build -ldflags "$(GOLDFLAGS)" ./...
 
+.PHONY: depscheck
+depscheck:
+	@echo "==> Checking source code dependencies..."
+	@go mod tidy
+	@git diff --exit-code -- go.mod go.sum || \
+		(echo; echo "Found differences in go.mod/go.sum files. Run 'go mod tidy' or revert go.mod/go.sum changes."; exit 1)
+	@# reset go.sum to state before checking if it is clean
+	@git checkout -q go.sum
+
+.PHONY: benchmark
 benchmark:
-	@echo "Running benchmarks..."
-	@$(foreach pkg,$(PACKAGES), \
-		go test -bench="." $(GO_BENCHMARK_FLAGS) $(pkg);)
+	go test -bench=. -benchmem ./...
+
+.PHONY: test
+test: tools
+	go test 				\
+		-race 				\
+		-coverprofile coverage.out 	\
+		./...
+	go tool cover -html=coverage.out -o coverage.html
+
+.PHONY: go-lint
+go-lint: tools
+	@echo "==> Checking source code against linters..."
+	@tools/golangci-lint run ./...
+	@tools/golangci-lint run --build-tags integration ./...
+
+.PHONY: docs-lint
+docs-lint: tools
```

Comment:
only now realized you didn't copy the `documentation` workflow over, which uses this target. I'd advise to copy that workflow, too.